services/datalad/datalad_service/tasks/publish.py

```diff
@@ -145,6 +149,43 @@ def publish_github_async(store, dataset, snapshot, github_remote):
     ds = store.get_dataset(dataset)
     publish_target(ds, github_remote, snapshot)
 
+def delete_s3_sibling(dataset_id, siblings, realm):
+    sibling = get_sibling_by_name(realm.s3_remote, siblings)
+    if sibling:
+        try:
+            session = boto3.Session(
+                aws_access_key_id=AWS_ACCESS_KEY_ID,
+                aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+            )
+            s3 = session.resource('s3')
+            bucket = s3.Bucket(realm.s3_bucket)
+            bucket.objects.filter(Prefix=f'{dataset_id}/').delete()
+        except Exception as e:
+            raise Exception(f'Attempt to delete dataset {dataset_id} from {realm.s3_remote} has failed. ({e})')
+
+def delete_github_sibling(dataset_id):
+    ses = Github(DATALAD_GITHUB_LOGIN, DATALAD_GITHUB_PASS)
+    org = ses.get_organization(DATALAD_GITHUB_ORG)
+    repos = org.get_repos()
+    try:
+        r = next(r for r in repos if r.name == dataset_id)
+        r.delete()
+    except StopIteration:
+        raise Exception(f'Attempt to delete dataset {dataset_id} from GitHub has failed, because the dataset does not exist.')
+
+def delete_siblings(store, dataset_id):
+    ds = store.get_dataset(dataset_id)
+
+    siblings = ds.siblings()
+    delete_s3_sibling(dataset_id, siblings, DatasetRealm(1))
+    delete_s3_sibling(dataset_id, siblings, DatasetRealm(2))
```

Comment:
These should use the enum names in case they are reordered or additional remote realms are defined later.